CMakeLists.txt

```diff
@@ -111,107 +155,96 @@ elseif(NOT WIN32)
+		"$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/external/crashpad>"
+		"$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/external/crashpad/third_party/mini_chromium/mini_chromium>"
+	)
+	target_compile_definitions(sentry PRIVATE SENTRY_WITH_CRASHPAD_BACKEND)
+	target_link_libraries(sentry PRIVATE crashpad_client crashpad_util)
+	if(NOT BUILD_SHARED_LIBS)
+		sentry_install(TARGETS crashpad_client crashpad_util crashpad_compat getopt mini_chromium zlib EXPORT sentry
+			LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+			ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+		)
+	endif()
 elseif(BACKEND_INPROC)
-	add_definitions(-DSENTRY_WITH_INPROC_BACKEND)
+	target_compile_definitions(sentry PRIVATE SENTRY_WITH_INPROC_BACKEND)
 endif()
 
-# ===== sentry library =====
-
-add_library("sentry" "${CMAKE_CURRENT_SOURCE_DIR}/vendor/mpack.c")
-add_subdirectory(src)
-
-set_target_properties("sentry" PROPERTIES PUBLIC_HEADER "include/sentry.h")
-target_link_libraries("sentry" ${LINK_LIBRARIES})
-install(TARGETS "sentry"
-	ARCHIVE DESTINATION lib
-	LIBRARY DESTINATION lib
-	PUBLIC_HEADER DESTINATION include)
-
-# https://gitlab.kitware.com/cmake/cmake/issues/18393
-if(BUILD_SHARED_LIBS)
-	if(APPLE)
-		install(FILES "$<TARGET_FILE:sentry>.dSYM" DESTINATION lib)
-	elseif(WIN32)
-		install(FILES $<TARGET_PDB_FILE:sentry> DESTINATION lib)
-	endif()
-endif()
+include(CMakePackageConfigHelpers)
+configure_package_config_file(sentry-config.cmake.in sentry-config.cmake
+	INSTALL_DESTINATION "${CMAKE_INSTALL_CMAKEDIR}")
 
-if(BACKEND_CRASHPAD)
-	add_dependencies("sentry" "crashpad_handler")
-endif()
+sentry_install(TARGETS sentry EXPORT sentry
+	ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+	LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+	RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
+	PUBLIC_HEADER DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
+)
+sentry_install(EXPORT sentry NAMESPACE sentry:: FILE sentry-targets.cmake
+	DESTINATION "${CMAKE_INSTALL_CMAKEDIR}")
+sentry_install(FILES "${PROJECT_BINARY_DIR}/sentry-config.cmake"
+	DESTINATION "${CMAKE_INSTALL_CMAKEDIR}")
 
 # ===== tests =====
 
 # compile tests separately and pass an extra preprocessor define so we can
 # switch some internal modes for the unittests.
+
 get_target_property(SENTRY_NATIVE_ALL_SOURCES sentry SOURCES)
-add_executable("sentry_test_unit" EXCLUDE_FROM_ALL
-	${SENTRY_NATIVE_ALL_SOURCES}
-	${SENTRY_NATIVE_TEST_SOURCES}
-)
-get_target_property(SENTRY_DEFS sentry COMPILE_DEFINITIONS)
-target_compile_definitions("sentry_test_unit" PUBLIC SENTRY_UNITTEST ${SENTRY_DEFS})
-target_link_libraries("sentry_test_unit" ${LINK_LIBRARIES})
+get_target_property(SENTRY_COMPILE_DEFINITIONS sentry COMPILE_DEFINITIONS)
+get_target_property(SENTRY_INTERFACE_INCLUDE_DIRECTORIES sentry INTERFACE_INCLUDE_DIRECTORIES)
+get_target_property(SENTRY_INCLUDE_DIRECTORIES sentry INCLUDE_DIRECTORIES)
+get_target_property(SENTRY_LINK_LIBRARIES sentry LINK_LIBRARIES)
+
+add_executable(sentry_test_unit EXCLUDE_FROM_ALL ${SENTRY_NATIVE_ALL_SOURCES})
```

Comment:
eventually make this optional e.g. add option and set it to `ON`. this allows consumers of this lib to not have to compile the tests...

same goes for the example

Reply:
good comment! I thought the `EXCLUDE_FROM_ALL` was enough. But cmake still processes all the files. We would ideally want to exclude the `./tests/` dir from the source file, but ran into cmake problems.

Reply:
Same for the example(s)?
SENTRY_BUILD_TESTS
SENTRY_BUILD_EXAMPLES

Reply:
I think that makes sense

Reply:
Done